Client/src/Components/Graph/AxesControl.tsx

```diff
@@ -0,0 +1,224 @@
+            }
+        })
+        console.log(missingDatasets)
+        setYVariableMissing(missingDatasets)
+    }
+
+    return (
+        <>
+            <Grid container direction='row'>
+                <Grid item xs={12}>
+                    <Button id='settingsToggle' variant="contained" onClick={handleSettingsClick} color="primary">
+                        Settings
+              {showSettings ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+                    </Button>
+                </Grid>
+                {showSettings &&
+                    <>
+                        <Paper elevation={10}>
+                            <Grid container direction='row' spacing={4}>
+                                <Grid item xs={4}>
+                                    <FormControl>
+                                        <InputLabel id="xVariable">X Variable</InputLabel>
+                                        <Select
+                                            labelId="xVariable"
+                                            id="xVariable"
+                                            value={props.axes.xVariableName}
+                                            autoWidth={true}
+                                            onChange={handleXVariableChange}
+                                        >
+                                            {props.variables.map(type => (
+                                                <MenuItem value={type}>{type}</MenuItem>
+                                            ))}
+                                        </Select>
+                                    </FormControl>
+                                </Grid>
+                                <Grid item xs={4}>
+                                    <FormControl>
+                                        <InputLabel id="xUnits">X Units</InputLabel>
+                                        <Select
+                                            labelId="xUnits"
+                                            id="xUnits"
+                                            value={props.axes.xVariableUnits}
+                                            autoWidth={true}
+                                            onChange={handleXUnitChange}
+                                        >
+                                            {xUnits.map(type => (
+                                                <MenuItem value={type}>{type}</MenuItem>
+                                            ))}
+                                        </Select>
+                                    </FormControl>
+                                </Grid>
+                                <Grid item xs={4}>
+                                    <Typography align="center">
+                                        Datasets Missing X: {xVariableMissing.toString()}
+                                    </Typography>
+                                </Grid>
```

Comment:
peeeeeerfect, thi sis what im talking about, very nice ,very easy for me to integrate, good job